ext/GraphNeuralNetworksSimpleWeightedGraphsExt/GNNGraphs/gnngraph.jl

```diff
@@ -0,0 +1,23 @@
+__precompile__(false)
+
+function GraphNeuralNetworks.GNNGraph(g::T; edge_weight = nothing, kws...) where 
+                 {T <: Union{SimpleWeightedGraph, SimpleWeightedDiGraph, 
+                       GraphNeuralNetworks.AbstractGraph}}
+    s = Graphs.src.(Graphs.edges(g))
+    t = Graphs.dst.(Graphs.edges(g))
+    if g isa Union{SimpleWeightedGraph, SimpleWeightedDiGraph}
+        w = filter(!iszero, g.weights |> vec) |> collect
```

Comment:
What is the need of the filtering? And why the collect?

Reply:
Also, if edge_weight != nothing we should use that instead

Reply:
Filtering is due to the format of SimpleWeightedGraph `g.weights` which include zero entries for not connected nodes (unlike GNN weights). 

Collect is to cast values to type expected by GNN graph weights.

`Also, if edge_weight != nothing we should use that instead`
I think no, this implementation is specific for SimpleWeightedGraphs weights. I assume creation of GNN graph with weights is already supported and this extension is aiming at adressing SimpleWeightedGraphs conversion specifically. I would leave other cases out of it, not to lengthen developer feedback loop when not necessary (I mean loading another package).

Reply:
```julia
julia> g.weights
3×3 SparseArrays.SparseMatrixCSC{Float64, Int64} with 6 stored entries:
  ⋅   0.5  2.0
 0.5   ⋅   0.8
 2.0  0.8   ⋅ 

julia> w = filter(!iszero, g.weights |> vec)
6-element SparseArrays.SparseVector{Float64, Int64} with 6 stored entries:
       ⋮
  [6]  =  0.8

julia> w = filter(!iszero, g.weights |> vec) |> collect
6-element Vector{Float64}:
 0.5
 2.0
 0.5
 0.8
 2.0
 0.8
```